stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/attachments/content/MediaAttachmentPreviewContent.kt

```diff
@@ -0,0 +1,116 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.compose.ui.attachments.content
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.unit.dp
+import com.getstream.sdk.chat.utils.extensions.imagePreviewUrl
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.compose.ui.components.CancelIcon
+import io.getstream.chat.android.compose.ui.components.composer.MessageInput
+import io.getstream.chat.android.compose.ui.util.rememberStreamImagePainter
+import io.getstream.chat.android.uiutils.constant.AttachmentType
+
+/**
+ * UI for currently selected image and video attachments, within the [MessageInput].
+ *
+ * @param attachments Selected attachments.
+ * @param onAttachmentRemoved Handler when the user removes an attachment from the list.
+ * @param modifier Modifier for styling.
+ */
+@Composable
+public fun MediaAttachmentPreviewContent(
+    attachments: List<Attachment>,
+    onAttachmentRemoved: (Attachment) -> Unit,
+    playButton: @Composable () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    LazyRow(
+        modifier = modifier.clip(RoundedCornerShape(topStart = 16.dp, topEnd = 16.dp)),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(4.dp, Alignment.Start)
+    ) {
+        items(attachments) { image ->
+            MediaAttachmentPreviewItem(
+                mediaAttachment = image,
+                onAttachmentRemoved = onAttachmentRemoved,
+                playButton = playButton
+            )
+        }
+    }
+}
+
+/**
+ * A preview of an individual selected image or video attachment.
+ *
+ * @param mediaAttachment The selected attachment.
+ * @param onAttachmentRemoved Handler when the user removes an attachment from the list.
+ * @param playButton Represents the play button that is overlaid above video attachment
+ * previews.
+ */
+@Composable
+private fun MediaAttachmentPreviewItem(
+    mediaAttachment: Attachment,
+    onAttachmentRemoved: (Attachment) -> Unit,
+    playButton: @Composable () -> Unit,
```

Comment:
and here

Reply:
Resolved using a similar strategy as here: https://github.com/GetStream/stream-chat-android/pull/4096/files#r965787775